Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs

```diff
@@ -1450,7 +1562,42 @@ public unsafe bool WasCompletedThisFrame()
             {
                 var actionStatePtr = &state.actionStates[m_ActionIndexInState];
                 var currentUpdateStep = InputUpdate.s_UpdateStepCount;
-                return actionStatePtr->lastCompletedInUpdate == currentUpdateStep && currentUpdateStep != default && actionStatePtr->frame == ExpectedFrame();
+                return actionStatePtr->lastCompletedInUpdate == currentUpdateStep && currentUpdateStep != default;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Check whether <see cref="phase"/> transitioned from <see cref="InputActionPhase.Performed"/> to any other phase
+        /// value at least once in the MonoBehaviour Update cycle (rendering frame).
+        /// </summary>
+        /// <returns>True if the action completed in this MonoBehaviour Update cycle (rendering frame).</returns>
+        /// <remarks>
+        /// Unlike <see cref="WasCompletedThisFrame"/>, this method will return true only if the InputSystem was updated and the action was completed in the current dynamic Update cycle (in between the previous and the current frame).
+        /// This can be used in dynamic update if the <see cref="InputSettings.updateMode"/> is set to <see cref="InputSettings.UpdateMode.ProcessEventsInFixedUpdate"/> or <see cref="InputSettings.UpdateMode.ProcessEventsManually"/>.
+        /// If the update mode is set to <see cref="InputSettings.UpdateMode.ProcessEventsInDynamicUpdate"/>, this method will behave exactly like <see cref="WasCompletedThisFrame"/>.
+        /// </remarks>
+        /// <example>
+        /// <code>
+        /// var warp = playerInput.actions["Warp"];
+        /// if (warp.WasPerformed())
+        ///     InitiateWarp();
```

Comment:
```suggestion
        /// var teleport = playerInput.actions["Teleport"];
        /// if (teleport.WasPerformedThisDynamicUpdate())
        ///     InitiateTeleport();
        /// else if (teleport.WasCompletedThisDynamicUpdate())
        ///     StopTeleport();
```
This code example was copied from the wrong method. This suggestion is based on `WasCompletedThisFrame`.